common/flatpak-dir.c

```diff
@@ -2127,10 +2131,30 @@ flatpak_dir_pull_extra_data (FlatpakDir          *self,
 
       /* TODO: Download to disk to support resumed downloads on error */
 
-      ensure_soup_session (self);
-      bytes = flatpak_load_http_uri (self->soup_session, extra_data_uri, NULL, NULL,
-                                     extra_data_progress_report, &extra_data_progress,
-                                     cancellable, error);
+      extra_local_file = flatpak_build_file (base_dir, "extra-data", extra_data_sha256, extra_data_name, NULL);
+      if (g_file_query_exists (extra_local_file, cancellable))
+        {
+          g_debug ("Loading extra-data from local file %s", g_file_get_path (extra_local_file));
+          gsize extra_local_size;
+          g_autofree char *extra_local_contents = NULL;
+          if (g_file_load_contents (extra_local_file, cancellable, &extra_local_contents, &extra_local_size, NULL, error) &&
+              extra_local_size == download_size)
+            {
+              bytes = g_bytes_new (extra_local_contents, extra_local_size);
+            }
+          else
+            {
+              return flatpak_fail (error, _("Wrong size for extra-data %s"), g_file_get_path (extra_local_file));
```

Comment:
This will double-set error if it was set by g_file_load_contents, which will g_error